### ISSUE TITLE
Integration Test Annotation magentoAppArea breaks with some valid values

### ISSUE THREAD

**Issue by Vinai** [magento/magento2#2907]
The following test breaks due to `Annotation\AppArea` trying to init the design.

``` php
/**
 * @magentoAppArea webapi_rest
 */
public function testTheModuleRegisteresAnAddressRepositoryPlugin()
{
    /** @var \Magento\TestFramework\Interception\PluginList $pluginList */
    $pluginList = $this->objectManager->get(PluginList::class);
    $pluginInfo = $pluginList->get(AddressRepositoryInterface::class, []);
    $this->assertSame(AddressRepositoryPlugin::class, $pluginInfo['example_module'[['instance']);
}
```

The relevant part of the stack trace is:

```
PHP Fatal error:  Uncaught InvalidArgumentException: Incorrect theme identification key in .../mage2-ce/vendor/magento/framework/View/Design/Theme/FlyweightFactory.php:56
Stack trace:
#0 .../mage2-ce/vendor/magento/module-theme/Model/View/Design.php(147): Magento\Framework\View\Design\Theme\FlyweightFactory->create(NULL, 'webapi_rest')
#1 .../mage2-ce/vendor/magento/module-theme/Model/View/Design.php(211): Magento\Theme\Model\View\Design->setDesignTheme(NULL)
Fatal error: Uncaught InvalidArgumentException: Incorrect theme identification key in .../mage2-ce/vendor/magento/framework/View/Design/Theme/FlyweightFactory.php on line 56
#2 .../mage2-ce/vendor/magento/framework/App/Area.php(256): Magento\Theme\Model\View\Design->setDefaultDesignTheme()
#3 .../mage2-ce/vendor/magento/framework/App/Area.php(214): Magento\Framework\App\Area->_initDesign()
InvalidArgumentException: Incorrect theme identification key in .../mage2-ce/vendor/magento/framework/View/Design/Theme/FlyweightFactory.php on line 56
#4 .../mage2-ce/vendor/magento/framework/App/Area.php(136): Magento\Framework\App\Area->_loadPart('design')

...

Magento\TestFramework\Annotation\AppArea->startTest() .../mage2-ce/dev/tests/integration/framework/Magento/TestFramework/EventManager.php:49
Magento\TestFramework\Application->loadArea() .../mage2-ce/dev/tests/integration/framework/Magento/TestFramework/Annotation/AppArea.php:82
Magento\TestFramework\Helper\Bootstrap->loadArea() .../mage2-ce/dev/tests/integration/framework/Magento/TestFramework/Application.php:615
Magento\Framework\App\Area->load() .../mage2-ce/dev/tests/integration/framework/Magento/TestFramework/Helper/Bootstrap.php:157
Magento\Framework\App\Area->_loadPart() .../mage2-ce/vendor/magento/framework/App/Area.php:136
Magento\Framework\App\Area->_initDesign() .../mage2-ce/vendor/magento/framework/App/Area.php:214
Magento\Theme\Model\View\Design->setDefaultDesignTheme() .../mage2-ce/vendor/magento/framework/App/Area.php:256
Magento\Theme\Model\View\Design->setDesignTheme() .../mage2-ce/vendor/magento/module-theme/Model/View/Design.php:211
Magento\Framework\View\Design\Theme\FlyweightFactory->create() .../mage2-ce/vendor/magento/module-theme/Model/View/Design.php:147
```

This is despite `\Magento\TestFramework\Annotation\AppArea` defining the valid areas as

``` php
/**
 * List of allowed areas
 *
 * @var array
 */
private $_allowedAreas = [
    \Magento\Framework\App\Area::AREA_GLOBAL,
    \Magento\Backend\App\Area\FrontNameResolver::AREA_CODE,
    \Magento\Framework\App\Area::AREA_FRONTEND,
    'webapi_rest',
    'webapi_soap',
    'cron',
];
```

The same error happens for `webapi_soap` and `cron`, too.
The other ones, `frontend`, `adminhtml` and `global` work as expected.

**Comment by Vinai** [comment]
In case google brings someone here, a possible workaround instead of using the `@magentoAppArea` annotation is:

``` php
public function testTheModuleRegisteresAnAddressRepositoryPlugin()
{
    /** @var ScopeInterface $scopeConfig */
    $scopeConfig = $this->objectManager->get(ScopeInterface::class);
    $scopeConfig->setCurrentScope('webapi_rest');

    /** @var \Magento\TestFramework\Interception\PluginList $pluginList */
    $pluginList = $this->objectManager->get(PluginList::class);
    $pluginInfo = $pluginList->get(AddressRepositoryInterface::class, []);
    $this->assertSame(AddressRepositoryPlugin::class, $pluginInfo['example_module'[['instance']);
}
```

Just be aware the setting will persist between tests, so be sure to reset the current scope in `tearDown` (something that the annotation does automatically).

**Comment by duhon** [comment]
Hello @Vinai 

Please, provide the used version. If the problem is actual for a specific tag, please, specify it and be sure that the latest update was used.

**Comment by Vinai** [comment]
This happens in `2.1.0-rc3` and the HEAD of the `develop` branch.

**Comment by mam08ixo** [comment]
Thanks @Vinai – Google brought me here. The issue still exists in `2.2.0-dev` (da064df).

**Comment by magento-engcom-team** [comment]
@Vinai, thank you for your report.
We've created internal ticket(s) MAGETWO-84466 to track progress on the issue.

**Comment by ishakhsuvarov** [comment]
@Vinai We are closing this ticket now as the issue seems to be fixed now in the `2.2-develop` and will be released with 2.2.4.
You may refer to https://github.com/magento/magento2/commit/506d5004e472f3937d4d67fefce62c9e2ce9c403 for more information on the delivered fix.

Thanks!

**Comment by magento-team** [comment]
Hi @Vinai. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#1194 by @nmalevanec in 2.3-develop branch
Related commit(s):
 - d7f13d03e2742319f93b366003a17eba3831295c

The fix will be available with the upcoming patch release.
